### PR TITLE
feat(api): Expose providers.yaml through server

### PR DESCRIPTION
**Pull request by nalanj**
## Describe your changes

See https://linear.app/nango/issue/NAN-2105/dynamically-reload-providersyaml

Adds an unsecured endpoint `/providers.json` that serves up the providers.yaml file data as JSON.

A future PR will add dynamic reloading to runners.

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is:
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/providers.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ function getProvidersPath() {
     return findProvidersYaml(dirname());
 }
 
-function loadProvidersYaml(): Record<string, Provider> | undefined {
+export function loadProvidersYaml(): Record<string, Provider> | undefined {
```
Comment (TBonnin) [inline packages/shared/lib/services/providers.ts]: does this need to be exported?

File: packages/server/lib/controllers/v1/getProvidersJSON.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { getProviders } from '@nangohq/shared';
+import { asyncWrapper } from '../../utils/asyncWrapper.js';
+
+let providersJSON: string | undefined = undefined;
+
+export const getProvidersJSON = asyncWrapper<any, any>((_, res) => {
+    if (!providersJSON) {
+        providersJSON = JSON.stringify(getProviders());
+    }
```
Comment (TBonnin) [inline packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: the providers content is already cached in memory. Is it worth caching the json as well? How slow/fast is the stringifying?

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/providers.ts (hunk 1 above)
Comment (nalanj) [reply packages/shared/lib/services/providers.ts]: Oh nope, that was from a previous idea of how to do this. Removing!

Comment (nalanj) [reply packages/shared/lib/services/providers.ts]: Fixed in cb23febc

File: packages/server/lib/controllers/v1/getProvidersJSON.ts (hunk 2 above)
Comment (nalanj) [reply packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: Do you mean caching in `shared` like we do with the providers data itself? This is caching the data in memory in the controller here. 

The reason I put this here is that I wasn't sure if we wanted to have that extra chunk of memory always in use by anything that uses providers from shared. It's currently about 190kb, so it's not like it's too big of a deal, but I figured we didn't want side effects that like that consistently leaking out of `shared`.

It takes around 500-700 microseconds to stringify the JSON right now and only has to happen the first time this controller is called, so I don't think it's too big of a deal.

Comment (TBonnin) [reply packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: up to you. It doesn't seem to be a big deal either way. 🚢

Comment (bodinsamuel) [reply packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: All for caching, but if you do it, you will need to deal with cache invalidation in the next PR 
Here you have no way to clear the cache easily

**Review by bodinsamuel — APPROVED** [review]

File: packages/server/lib/controllers/v1/getProvidersJSON.ts (hunk 2 above)
Comment (TBonnin) [reply packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: server restart/deployment would "invalidate" the cache.

Comment (bodinsamuel) [reply packages/server/lib/controllers/v1/getProvidersJSON.ts:9]: ah yes true we rely on the deploy, then fine 👌🏻

**Pull request merged**